### PR TITLE
Added Clone method

### PR DESCRIPTION
**Pull request by bdeining**

### CODE REVIEW

File: src/main/java/com/connexta/ddf/catalog/direct/CatalogMethods.java [hunk 1]
```diff
@@ -91,7 +99,7 @@
   {
     Builder<String, DocMethod> builder = ImmutableMap.builder();
     builder.put(
-        "ddf.catalog/create",
+        CREATE_KEY,
```
Comment (gjvera) [inline src/main/java/com/connexta/ddf/catalog/direct/CatalogMethods.java]: Any particular reason this key was moved to it's own variable, but not the other methods?

File: src/main/java/com/connexta/ddf/catalog/direct/CatalogMethods.java [hunk 2]
```diff
@@ -123,6 +131,8 @@
             this::delete,
             "Takes the specified parameters and calls CatalogFramework::query. `params` takes:"
                 + " `ids` (Required, value: List(String))"));
+
+    builder.put("ddf.catalog/clone", new DocMethod(this::clone, ""));
```
Comment (Bdthomson) [inline src/main/java/com/connexta/ddf/catalog/direct/CatalogMethods.java]: ✏️ Could add documentation for the params this takes, similar to the delete call above.

File: src/main/java/com/connexta/ddf/catalog/direct/CatalogMethods.java [hunk 3]
```diff
@@ -183,6 +193,62 @@ private Object getSourceInfo(Map<String, Object> params) {
...
+    metacardToClone.setAttribute(new AttributeImpl(Core.ID, (Serializable) null));
+    metacardToClone.setAttribute(
+        new AttributeImpl(Security.ACCESS_ADMINISTRATORS, (Serializable) null));
+    metacardToClone.setAttribute(new AttributeImpl(Security.ACCESS_GROUPS, (Serializable) null));
+    metacardToClone.setAttribute(
+        new AttributeImpl(Security.ACCESS_GROUPS_READ, (Serializable) null));
+    metacardToClone.setAttribute(
+        new AttributeImpl(Security.ACCESS_INDIVIDUALS, (Serializable) null));
+    metacardToClone.setAttribute(
+        new AttributeImpl(Security.ACCESS_INDIVIDUALS_READ, (Serializable) null));
```
Comment (Bdthomson) [inline src/main/java/com/connexta/ddf/catalog/direct/CatalogMethods.java]:
```suggestion
clearAttributes(Metacard metacard, List<Attribute> attributesToClear){
  attributesToClear.forEach(attributeToClear -> {
    metacard.setAttribute(attributeToClear, (Serializable) null)
  })
}

clearAttributes(metacardToClone, new ArrayList<Attribute>{
Security.ACCESS_ADMINISTRATORS,
Security.ACCESS_GROUPS,
...
})
```

File: src/main/java/com/connexta/ddf/catalog/direct/CatalogMethods.java (hunk 1 above)
Comment (bdeining) [reply src/main/java/com/connexta/ddf/catalog/direct/CatalogMethods.java]: not really, i can revert it if we don't want it there, but ideally they all are extracted out to constants when the unit tests are written.

**Comment by willwill96** [comment]
:question: Will the catalog framework handle attributes like `created` and `modified`?

File: src/main/java/com/connexta/ddf/catalog/direct/CatalogMethods.java [hunk 4]
```diff
@@ -183,6 +193,62 @@ private Object getSourceInfo(Map<String, Object> params) {
...
+    Filter filter = filterBuilder.allOf(idFilter, tagFilter);
+    Query query = new QueryImpl(filter);
+
+    QueryRequest queryRequest = new QueryRequestImpl(query);
+
+    QueryResponse queryResponse;
+    try {
+      queryResponse = catalogFramework.query(queryRequest);
+    } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+      return new Error(INTERNAL_ERROR, e.getMessage());
```
Comment (mojogitoverhere) [inline src/main/java/com/connexta/ddf/catalog/direct/CatalogMethods.java]: Should these exceptions also be logged?

File: src/main/java/com/connexta/ddf/catalog/direct/CatalogMethods.java [hunk 5]
```diff
@@ -123,6 +132,12 @@
             this::delete,
             "Takes the specified parameters and calls CatalogFramework::query. `params` takes:"
                 + " `ids` (Required, value: List(String))"));
+
+    builder.put(
+        "ddf.catalog/clone",
```
Comment (pklinef) [inline src/main/java/com/connexta/ddf/catalog/direct/CatalogMethods.java]: :exclamation: The goal of the CatalogMethods JSON RPC API was to mirror the Catalog Framework API as close as possible.  I think this clone method belongs in a GraphQL data resolver or in a new JSON RPC endpoint instead.

**Comment by bdeining** [comment]
https://github.com/connexta/ddf-jsonrpc/pull/11#issuecomment-592094538
created / modified are now cleared during the clone

File: src/main/java/com/connexta/ddf/catalog/direct/ExtendedMethods.java [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+import ddf.catalog.source.UnsupportedQueryException;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.opengis.filter.Filter;
+
+public class ExtendedMethods implements MethodSet {
```
Comment (pklinef) [inline src/main/java/com/connexta/ddf/catalog/direct/ExtendedMethods.java]: :pencil2: Could add JavaDoc to explain this is for methods that extend the catalog framework methods.

File: src/main/java/com/connexta/ddf/catalog/direct/Utils.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.AttributeType.AttributeFormat;
+import ddf.catalog.data.Metacard;
+import java.nio.charset.Charset;
+import java.util.Base64;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class Utils {
```
Comment (pklinef) [inline src/main/java/com/connexta/ddf/catalog/direct/Utils.java]: :pencil2: Could rename this to `MetacardMap` and make the method `convert`.  Or the class to `MetacardTransform` and the method to `toMap` if we expect to have other types of metacard transformations.

**Pull request merged**